### PR TITLE
Polish `DefaultSetProperty`

### PR DESCRIPTION
**Pull request by bamboo**
- Replace anonymous inner class by lambda

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle merge

**Comment by bamboo** [comment]
Thanks, @abstratt, please look again.

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Comment by bamboo** [comment]
@bot-gradle merge

File: subprojects/core-api/src/main/java/org/gradle/api/internal/lambdas/SerializableLambdas.java [hunk 1]
```diff
@@ -56,6 +57,10 @@ public static <T, U, R> BiFunction<T, U, R> bifunction(SerializableBiFunction<T,
         return f;
     }
 
+    public static <T> Supplier<T> spec(SerializableSupplier<T> supplier) {
```
Comment (mlopatkin) [inline subprojects/core-api/src/main/java/org/gradle/api/internal/lambdas/SerializableLambdas.java:60]: :x: Should this method be called `spec`? I'd prefer `supplier`.

Comment (bamboo) [reply subprojects/core-api/src/main/java/org/gradle/api/internal/lambdas/SerializableLambdas.java:60]: 😮

**Comment by bamboo** [comment]
@bot-gradle cancel

File: subprojects/core-api/src/main/java/org/gradle/api/internal/lambdas/SerializableLambdas.java (hunk 1 above)
Comment (bamboo) [reply subprojects/core-api/src/main/java/org/gradle/api/internal/lambdas/SerializableLambdas.java]: Thanks, Mike, crisis adverted! 😄

**Comment by bamboo** [comment]
@bot-gradle merge

**Pull request merged**